samples/javascript_typescript/60.multilingual-luis-bot/translator.ts

```diff
@@ -0,0 +1,345 @@
+        this.postProcessor = new TranslatorPostProcessor(noTranslatePatterns, wordDictionary);
+
+    }
+
+    public detect = async (text: string): Promise<object> => {
+        let uri: any = "https://api.cognitive.microsofttranslator.com/detect?api-version=3.0";
+        let translationRequest = new TranslationRequest();
+        translationRequest.Text = text;
+        const response = await request({
+            url: uri,
+            method: 'POST',
+            headers: {
+                'Ocp-Apim-Subscription-Key': this.apiKey,
+                'Content-Type': 'application/json',
+            },
+            body: JSON.stringify([translationRequest]),
+        });
+        let detectedLanguage = JSON.parse(response);
+        return detectedLanguage;
+    };
+
+    public translateArray = async (options: TranslateArrayOptions):  Promise<object> => {
+		try {
+            let from = options.from;
+            let to = options.to;
+            let texts = options.texts;
+            let orgTexts = [];
+            let translationRequests = [];
+            let results = [];
+            texts.forEach((text, index, array) => {
+                let normalizedText = text.trim();
+                    orgTexts.push(normalizedText);
+                    texts[index] = normalizedText;
+                    let translationRequest = new TranslationRequest();
+                    translationRequest.Text = normalizedText;
+                    translationRequests.push(translationRequest);
+            });
+            let uri: any = `https://api.cognitive.microsofttranslator.com/translate?api-version=3.0&includeAlignment=true&includeSentenceLength=true&from=${from}&to=${to}`;
+			const response = await request({
+				url: uri,
+				method: 'POST',
+				headers: {
+                    'Ocp-Apim-Subscription-Key': this.apiKey,
+					'Content-Type': 'application/json',
+				},
+				body: JSON.stringify(translationRequests),
+            });
+            let translations = JSON.parse(response);
+            translations.forEach((traslationModel, index, array) => {
+                                let translation = traslationModel.translations[0].text;
+                                if(traslationModel.translations[0].alignment){
+                                    let alignment = traslationModel.translations[0].alignment.proj;
+                                    translation = this.postProcessor.fixTranslation(orgTexts[index], alignment, translation, from);
+                                }
+                                let result: TranslationResult = { translatedText: translation };
+                                 results.push(result);
+                            });
+			return results;
+		} catch (e) {
+			switch (e.statusCode) {
+				case 401:
+					throw new Error('Cognitive Authentication Failed');
+				default:
+					throw new Error('Internal Error');
+			}
+		}
+    };
+}
+
+
+/**
+ * @private
```

Comment:
Are you that this class is private or is it just a typo?

Reply:
it's private